### DISCUSSION TITLE
Cyclic loading simulation of equal strain amplitude

### DISCUSSION THREAD

**Discussion by Ming-is** [neperfepx/FEPX#48] [Q&A] [answered]
I was trying to use fepx to simulate the cyclic loading test of controlled strain amplitude, but it failed. 
Here are the instructions and simulation file I us:
```
neper -T -n 100 -dim 3 -ori "file(ori,des=euler-bunge)" -oridescriptor "euler-bunge:active" -domain "cube(0.005,0.005,0.005)"  -statcell diameq  -for tess,tesr -o simulation
neper -M simulation.tess -order 2
```

The simulation.config file is set as follows：

```
## Optional Input
hard_type cyclic_isotropic
## Material Parameters
number_of_phases 1 
phase 1

    crystal_type FCC
    m 0.020d0
    gammadot_0 1.0d0
    h_0 375.0d0
    g_0 160.0d0
    g_s0 1000.0d0
    n 1.0d0
    c11 85.678e3
    c12 61.859e3
    c44 38.298e3
    cyclic_a 0.050d0
    cyclic_c 3.50d0

## Deformation History
def_control_by uniaxial_strain_target
number_of_strain_steps 3
target_strain  0.007 10 print_data
target_strain  0.00 10 print_data
target_strain  -0.007 10 print_data

## Boundary Conditions
boundary_conditions uniaxial_grip
loading_direction z
loading_face z1
strain_rate 1e-2

## Printing Results
print stress
print strain-eq
print forces
```

Problem：The above instructions run normally with neper，However, when I use  `mpirun -n 4 fepx` to run the simulation, the following error messages will appear:
```
mpirun noticed that process rank 2 with PID 0 on node ubuntu exited on signal 9 (Killed).
```
when I use  `fepx` to run the simulation, The program never seems to converge，the following error messages will appear：
```
Update hardening iteration limit reached.
Killed
```

So how do I need to modify it in order to simulate the cyclic loading of equal strain amplitude smoothly
[simulation.zip](https://github.com/acmelab-ua/FEPX/files/7539003/simulation.zip)
thanks

**Comment by mkasemer** [comment, marked as answer]
@Ming-is 

First, do you get the MPI error message with every simulation, including the examples?

Second, can you comment on all of your units for both your domain and your model parameters?

**Reply by Ming-is** [reply]
Running the reference example was successful .

cube(0.005,0.005 ,0.005)    -------mm
c11 85.678e3    ---------MPa
c12 61.859e3    ---------MPa
c44 38.298e3    ---------MPa             （I set other material parameters following the  reference example 4）

**Reply by Ming-is** [reply]
I have tried to combine my own congfig file with the tess and msh files of the example. Running ‘’mpirun -n 4 fepx ’‘can run smoothly. So I suspect that the tess and msh files generated by me are incompatible.

**Reply by Ming-is** [reply]
My neper is already version 4.2.0. When I tried to generate the Tess file of  the fepx  example 5 , I found that neper has been running for nearly 8 hours, but it has not been completed，At present, the computer has been running for 12 hours, and the following two instructions have not been completed successfully.The instructions are as follows：

neper -T \
    -n 50 \
    -reg 1 \
    -group "id<=25?1:2" \
    -ori "file(orifile,des=rodrigues)" \
    -oridistrib "normal(5)" \
    -oridescriptor "rodrigues:active" \
    -morpho gg \
    -o simulation

neper -T -n 100 -dim 3 -reg 1 -ori "file(ori,des=euler-bunge)" -oridescriptor "euler-bunge:active" -oridistrib "normal(5)" -morpho gg  -domain "cube(0.005,0.005,0.005)"  -statcell diameq  -for tess,tesr -o simulation


The confusion of fepx and neper made me wonder where the mistake happened for a while

**Reply by mkasemer** [reply]
Hi @Ming-is,

I will have a chance tomorrow to look at your specific case above. Regarding your most recent issue with Neper running for a very long time: if you have an issue with Neper, you should post in the Neper discussion forums.

**Reply by Ming-is** [reply]
I solved neper's problem.I think I know the reason. It should be that Ubuntu has been started for too long. After restarting the virtual machine, the program runs fast.But for the trouble of fepx ，i‘m looking forward to your solution

**Comment by Ming-is** [comment]
At present, fepx can run roughly by improving Tess and MSH files, but the following warnings will appear frequently in some incremental steps of the running process:
Warning:       . Update hardening iteration limit reached.
Warning:       . Not all crystals converged.

Although the final simulation can be completed, these non convergent incremental steps will greatly slow down the running speed of the program. What improvements do I need to make, such as the adjustment of material parameters?

**Reply by mkasemer** [reply]
@Ming-is: I'm glad you have FEPX working now. The warnings you describe are common when time steps are too large for the given parameters / domain. Try reducing time steps to get better convergence.
